rust/worker/src/execution/operators/hnsw_knn.rs

```diff
@@ -53,25 +54,24 @@ impl ChromaError for HnswKnnOperatorError {
             HnswKnnOperatorError::RecordSegmentError => ErrorCodes::Internal,
             HnswKnnOperatorError::RecordSegmentReadError => ErrorCodes::Internal,
             HnswKnnOperatorError::InvalidAllowedAndDisallowedIds => ErrorCodes::InvalidArgument,
+            HnswKnnOperatorError::LogMaterializationError(e) => e.code(),
         }
     }
 }
 
 impl HnswKnnOperator {
-    async fn get_disallowed_ids(
+    async fn get_disallowed_ids<'referred_data>(
         &self,
-        logs: Chunk<LogRecord>,
+        logs: Chunk<MaterializedLogRecord<'_>>,
         record_segment_reader: &RecordSegmentReader<'_>,
     ) -> Result<Vec<u32>, Box<dyn ChromaError>> {
         let mut disallowed_ids = Vec::new();
         for item in logs.iter() {
             let log = item.0;
-            let operation_record = &log.record;
-            if operation_record.operation == Operation::Delete
-                || operation_record.operation == Operation::Update
+            if log.final_operation == Operation::Delete || log.final_operation == Operation::Update
```

Comment:
I think we should comment the behavior that if an embedding does not get updated, we brute force it, even thought **we could** search for it in the hnsw index